sql/core/src/main/scala/org/apache/spark/sql/SQLConf.scala

```diff
@@ -449,17 +449,13 @@ private[spark] object SQLConf {
     doc = "When true, we could use `datasource`.`path` as table in SQL query"
   )
 
-  val SPECIALIZE_SINGLE_DISTINCT_AGG_PLANNING =
-    booleanConf("spark.sql.specializeSingleDistinctAggPlanning",
-      defaultValue = Some(true),
+  val AGG_PLANNING_15 =
+    booleanConf("spark.sql.aggregationPlanning15",
```

Comment:
But we are doing the same thing as maintaining compatibility and no (much) regressions, otherwise we should just remove this flag (it's not public right now).